apps/web/src/components/transactions/TxDetails/TxData/SpendingLimits/index.tsx

```diff
@@ -14,20 +14,24 @@ import type { SpendingLimitMethods } from '@/utils/transaction-guards'
 import { isSetAllowance } from '@/utils/transaction-guards'
 import chains from '@/config/chains'
 import TxDetailsRow from '@/components/tx/ConfirmTxDetails/TxDetailsRow'
+import useAddressBook from '@/hooks/useAddressBook'
+import { ZERO_ADDRESS } from '@safe-global/protocol-kit/dist/src/utils/constants'
 
 type SpendingLimitsProps = {
   txData?: TransactionData | null
   txInfo: CustomTransactionInfo
   type: SpendingLimitMethods
 }
 
-export const SpendingLimits = ({ txData, txInfo, type }: SpendingLimitsProps): ReactElement | null => {
+export const SpendingLimits = ({ txData, type }: SpendingLimitsProps): ReactElement | null => {
   const chain = useCurrentChain()
   const tokens = useAppSelector(selectTokens)
+  const addressBook = useAddressBook()
   const isSetAllowanceMethod = useMemo(() => isSetAllowance(type), [type])
 
   const [beneficiary, tokenAddress, amount, resetTimeMin] =
     txData?.dataDecoded?.parameters?.map(({ value }) => value) || []
+  const beneficiaryName = addressBook[beneficiary as string]
```

Comment:
why do we need todo this? If you look at the source code of EthHashInfo you'll see that we do an useAddressBookItem call there. Why is it not working?